src/nvim/highlight_group.c

```diff
@@ -230,60 +230,100 @@ static const char *highlight_init_both[] = {
+  "default link @string.documentation   Comment",
+  "default link @string.regexp          SpecialChar",
+  "default link @string.escape          SpecialChar",
+  "default link @string.special         SpecialChar",
+  "default link @string.special.symbol  Constant",
+  "default link @string.special.url     Underlined",
+
+  "default link @character          Character",
+  "default link @character.special  SpecialChar",
+
+  "default link @boolean       Boolean",
+  "default link @number        Number",
+  "default link @number.float  Float",
+
+  "default link @type             Type",
+  "default link @type.definition  Typedef",
+  "default link @type.qualifier   StorageClass",
+
+  "default link @attribute  Function",
+  "default link @property   Identifier",
+
+  "default link @function          Function",
+  "default link @function.builtin  Special",
+  "default link @function.macro    Macro",
+
+  "default link @constructor   Special",
+  "default link @operator      Operator",
+
+  "default link @keyword            Keyword",
+  "default link @keyword.function   Statement",
+  "default link @keyword.operator   Operator",
+  "default link @keyword.import     Include",
+  "default link @keyword.storage    StorageClass",
+  "default link @keyword.repeat     Repeat",
+  "default link @keyword.debug      Debug",
+  "default link @keyword.exception  Exception",
+
+  "default link @keyword.conditional    Conditional",
+
+  "default link @keyword.directive         Preproc",
+  "default link @keyword.directive.define  Define",
+
+  "default link @punctuation.delimiter  Delimiter",
+  "default link @punctuation.bracket    Delimiter",
+  "default link @punctuation.special    Special",
+
+  "default link @comment   Comment",
+
+  "default link @comment.error    DiagnosticError",
+  "default link @comment.warning  DiagnosticWarn",
+  "default link @comment.hint     DiagnosticHint",
+  "default link @comment.info     DiagnosticInfo",
+  "default link @comment.todo     Todo",
+
+  "@markup.strong         gui=bold           cterm=bold",
+  "@markup.italic         gui=italic         cterm=italic",
+  "@markup.strikethrough  gui=strikethrough, cterm=strikethrough",
+  "@markup.underline      gui=underline,     cterm=underline",
+
+  "default link @markup.heading  Title",
+
+  "default link @markup.quote        Comment",
+  "default link @markup.math         Identifier",
+  "default link @markup.environment  Identifier",
+
+  "default link @markup.link        Underlined",
+  "default link @markup.link.label  Identifier",
+  "default link @markup.link.url    Underlined",
+
+  "default link @markup.raw  Comment",
+
+  "default link @markup.list            Special",
+  "default link @markup.list.checked    DiagnosticOk",
+  "default link @markup.list.unchecked  DiagnosticWarn",
+
+  "default link @diff.plus   Added",
+  "default link @diff.minus  Removed",
+  "default link @diff.delta  Changed",
+
+  "default link @tag  Tag",
```

Comment:
I've seen couple of times that for people working with frontend having `@tag.delimiter` be different to what is inside tag is important for usability. So maybe add `@tag.delimiter` that links to `Delimiter`? It will be less noticeable with default color scheme (`Delimiter` is cleared for now), but semantically might be better?